dusty/source.py

```diff
@@ -96,11 +97,15 @@ def vm_path(self):
 
     @property
     def rel_path(self):
-        local_folder = self.remote_path
-        if ':' in local_folder:
-            local_folder = local_folder.split(':')[1]
-        local_folder = local_folder.lstrip('/')
-        return local_folder
+        parsed = urlparse.urlparse(self.remote_path.lstrip('/'))
+        if parsed.hostname:
+            local_folder = parsed.hostname + parsed.path
+        else:
+            local_folder = parsed.path
+        end_of_path = local_folder.split('/')[-1]
+        if '.' in end_of_path:
+            local_folder = local_folder.replace(end_of_path, end_of_path.split('.')[0])
```

Comment:
Crap, I need to make this work for paths that end in, to use an unfortunate GC example, `push.js.git`